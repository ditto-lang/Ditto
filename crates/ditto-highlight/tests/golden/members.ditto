module Test exports (..);


type Maybe(a) =
    | Nothing
    | Just(a);

nothing = Nothing;

is_just = fn (mb) ->
    match mb with
    | Maybe.Just(_) -> true
    | Nothing -> false
    end;
