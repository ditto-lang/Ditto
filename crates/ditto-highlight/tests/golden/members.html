<KeywordImport>module</KeywordImport> <Namespace>Test</Namespace> <KeywordImport>exports</KeywordImport> <Bracket>(</Bracket><Symbol>..</Symbol><Bracket>)</Bracket><Delimiter>;</Delimiter>


<Keyword>type</Keyword> <Type>Maybe</Type><Bracket>(</Bracket><TypeVariable>a</TypeVariable><Bracket>)</Bracket> =
    <Delimiter>|</Delimiter> <EnumMember>Nothing</EnumMember>
    <Delimiter>|</Delimiter> <EnumMember>Just</EnumMember><Bracket>(</Bracket><TypeVariable>a</TypeVariable><Bracket>)</Bracket><Delimiter>;</Delimiter>

<TopLevelName>nothing</TopLevelName> = <EnumMember>Nothing</EnumMember><Delimiter>;</Delimiter>

<TopLevelName>is_just</TopLevelName> = <Keyword>fn</Keyword> <Bracket>(</Bracket><Variable>mb</Variable><Bracket>)</Bracket> <Operator>-></Operator>
    <KeywordConditional>match</KeywordConditional> <Variable>mb</Variable> <KeywordConditional>with</KeywordConditional>
    <Delimiter>|</Delimiter> <Namespace>Maybe.</Namespace><EnumMember>Just</EnumMember><Bracket>(</Bracket>_<Bracket>)</Bracket> <Operator>-></Operator> <Boolean>true</Boolean>
    <Delimiter>|</Delimiter> <EnumMember>Nothing</EnumMember> <Operator>-></Operator> <Boolean>false</Boolean>
    <KeywordConditional>end</KeywordConditional><Delimiter>;</Delimiter>
